rskj-core/src/main/java/org/ethereum/datasource/DataSourceWithCache.java

```diff
@@ -56,14 +65,21 @@ public DataSourceWithCache(@Nonnull KeyValueDataSource base, int cacheSize) {
     }
 
     public DataSourceWithCache(@Nonnull KeyValueDataSource base, int cacheSize,
-                               @Nullable CacheSnapshotHandler cacheSnapshotHandler) {
+                               @Nullable CacheSnapshotHandler cacheSnapshotHandler,
+                               boolean preventWritesToBase) {
+        this.preventWritesToBase = preventWritesToBase;
         this.cacheSize = cacheSize;
         this.base = Objects.requireNonNull(base);
-        this.uncommittedCache = new LinkedHashMap<>(cacheSize / 8, (float) 0.75, false);
+        this.uncommittedCache = new HashMap<>();
```

Comment:
What is the rationale for this change? Is order not relevant?

Asking because there is a test that checks for insertion order and randomly fails after this change: `org.ethereum.datasource.DataSourceWithCacheTest#putTwoKeyValuesWrittenInOrder`

Reply:
Order is irrelevant.
See https://iov-labs.slack.com/archives/C747J6HN3/p1660151397008399 for a discussion

Reply:
Thanks. Then I will update the test.